go-controller/pkg/kubevirt/pod_test.go

```diff
@@ -0,0 +1,218 @@
+package kubevirt
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/rand"
+	kubevirtv1 "kubevirt.io/api/core/v1"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/factory"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+)
+
+const vmName = "test-vm"
+
+var _ = Describe("Kubevirt Pod", func() {
+	const (
+		t0 = time.Duration(0)
+		t1 = time.Duration(1)
+		t2 = time.Duration(2)
+		t3 = time.Duration(3)
+		t4 = time.Duration(4)
+	)
+	runningKvSourcePod := runningKubevirtPod(t0)
+	postSuccessfulMigrationKvSourcePod := completedKubevirtPod(t1)
+
+	failedMigrationKvTargetPod := failedKubevirtPod(t2)
+	successfulMigrationKvTargetPod := runningKubevirtPod(t3)
+	anotherFailedMigrationKvTargetPod := failedKubevirtPod(t4)
+	duringMigrationKvTargetPod := runningKubevirtPod(t4)
+	yetAnotherDuringMigrationKvTargetPod := runningKubevirtPod(t4)
+	readyMigrationKvTargetPod := domainReadyKubevirtPod(t4)
+
+	type testParams struct {
+		pods                    []corev1.Pod
+		expectedError           error
+		expectedMigrationStatus *LiveMigrationStatus
+	}
+	DescribeTable("DiscoverLiveMigrationStatus", func(params testParams) {
+		Expect(config.PrepareTestConfig()).To(Succeed())
+		config.OVNKubernetesFeature.EnableNetworkSegmentation = true
+		config.OVNKubernetesFeature.EnableMultiNetwork = true
+		config.OVNKubernetesFeature.EnableInterconnect = true
+
+		fakeClient := util.GetOVNClientset().GetOVNKubeControllerClientset()
+		wf, err := factory.NewOVNKubeControllerWatchFactory(fakeClient)
+		Expect(err).ToNot(HaveOccurred())
+
+		for _, pod := range params.pods {
+			_, err := fakeClient.KubeClient.CoreV1().Pods(pod.Namespace).Create(context.Background(), &pod, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+		}
+
+		Expect(wf.Start()).To(Succeed())
+		defer wf.Shutdown()
+
+		currentPod := params.pods[0]
+		migrationStatus, err := DiscoverLiveMigrationStatus(wf, &currentPod)
+		if params.expectedError == nil {
+			Expect(err).To(BeNil())
+		} else {
+			Expect(err.Error()).To(ContainSubstring(params.expectedError.Error()))
```

Comment:
nit: can't you ?
```suggestion
			Expect(err).To(MatchError(params.expectedError))
```

If not, please take into account that `MatchError` can receive matchers as input - i.e. `MatchError(ContainSubstring(...))` is legit.

Reply:
was not aware of that option! will fix to
```
Expect(err).To(MatchError(ContainSubstring(params.expectedError.Error())))
```

Reply:
good. But I still don't understand why you can't match the error directly.

Reply:
Once of the errors is pod name specific:
```
func tooManyPodsError(livingPods []*corev1.Pod) error {
	var podNames = make([]string, len(livingPods))
	for i := range livingPods {
		podNames[i] = livingPods[i].Namespace + "/" + livingPods[i].Name
	}
	return fmt.Errorf("unexpected live migration state at pods: %s", strings.Join(podNames, ","))
}
```
So I want to compare only the static part. AFAIK that's the best way to do it.